WiseBuy/APIs/UPCDatabaseAPI.m

```diff
@@ -0,0 +1,46 @@
+//
+//  UPCDatabaseAPI.m
+//  WiseBuy
+//
+//  Created by Mai Ngo on 7/21/22.
+//
+
+#import "UPCDatabaseAPI.h"
+
+static NSString *const kBaseURL = @"https://api.upcitemdb.com/prod/v1/lookup?upc=";
+
+@implementation UPCDatabaseAPI
+
++ (void)fetchDeals:(NSString *)barcode {
+    
+    NSString *const fullBaseURL = [NSString stringWithFormat:@"%@%@", kBaseURL, barcode];
+    
+    NSString *const path = [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"];
+    NSDictionary *const dict = [NSDictionary dictionaryWithContentsOfFile: path];
+    NSString *userKey = [dict objectForKey: @"searchUPC_userKey"];
+    
+    if ([[NSUserDefaults standardUserDefaults] stringForKey:@"searchUPC_userKey"]) {
+        userKey = [[NSUserDefaults standardUserDefaults] stringForKey:@"searchUPC_userKey"];
+    }
+    NSDictionary *headers = @{
+        @"user_key": userKey
+    };
+    
+    NSDictionary *responseDictionary = [APIHelper getResponseFromAPI:fullBaseURL : headers];
+    if ([responseDictionary[@"items"] count]) {
+        NSInteger count = [responseDictionary[@"items"][0][@"offers"] count];
+        
+        if (count > 0) {
+            Item *newItem = [APIHelper createItem:responseDictionary[@"items"][0][@"title"] :responseDictionary[@"items"][0][@"description"] :barcode :responseDictionary[@"items"][0][@"images"][0]];
```

Comment:
suggestion: If you are accessing a particular element of a dictionary frequently, it's recommended to store it as a local variable opposed to refetching it every time (thinking of responseDictionary[@"items"][0] in this case)